extensions/stats/testdata/istio/metadata-exchange_filter.yaml

```diff
@@ -3,43 +3,21 @@ kind: EnvoyFilter
 metadata:
   name: metadata-exchange
 spec:
-  filters:
-  - filterConfig:
-      configuration: envoy.wasm.metadata_exchange
-      vm_config:
-        code:
-          inline_string: envoy.wasm.metadata_exchange
-        vm: envoy.wasm.vm.null
-    filterName: envoy.wasm
-    filterType: HTTP
-    insertPosition:
-      index: FIRST
-    listenerMatch:
-      listenerProtocol: HTTP
-      listenerType: SIDECAR_INBOUND
-  - filterConfig:
-      configuration: envoy.wasm.metadata_exchange
-      vm_config:
-        code:
-          inline_string: envoy.wasm.metadata_exchange
-        vm: envoy.wasm.vm.null
-    filterName: envoy.wasm
-    filterType: HTTP
-    insertPosition:
-      index: FIRST
-    listenerMatch:
-      listenerProtocol: HTTP
-      listenerType: SIDECAR_OUTBOUND
-  - filterConfig:
-      configuration: envoy.wasm.metadata_exchange
-      vm_config:
-        code:
-          inline_string: envoy.wasm.metadata_exchange
-        vm: envoy.wasm.vm.null
-    filterName: envoy.wasm
-    filterType: HTTP
-    insertPosition:
-      index: FIRST
-    listenerMatch:
-      listenerProtocol: HTTP
-      listenerType: GATEWAY
+  configPatches:
+    - applyTo: HTTP_FILTER
+      match:
+        context: ANY # inbound, outbound, and gateway
+        listener:
+          filterChain:
+            filter:
+              name: "envoy.http_connection_manager"
+      patch:
+        operation: INSERT_BEFORE
```

Comment:
insert before what? missing sub filter match ?

Reply:
From the docs: "If no filter is selected, the specified filter will be inserted at the front of the list."

I believe that is what is desired.